semgrep/rules/python/filesystem/path_traversal.py

```diff
@@ -0,0 +1,70 @@
+import json
+import os
+
+from bs4 import UnicodeDammit
+
+
+def handle_request_1(request):
+    payload = json.loads(request.POST.get('payload'))
+
+    # ruleid: path-traversal
+    with open(payload) as f:
+        f.write('foo')
+
+
+def handle_request_2(request):
+    payload = json.loads(request.POST.get('payload'))
+    a = payload.get('a')
```

Comment:
would this rule also flag `payload['a']` (the other common form of dictionary access)?

Reply:
yep